src/pages/ProductDetail/ProductDetail.js

```diff
@@ -29,24 +49,91 @@ const ProductDetail = () => {
     }
     return 0;
   };
-
   const renderNumber = renderInterestedNum(productInterestedNum);
 
-  const favoriteModalClick = () => {
-    setFavoriteModal(true);
+  const clickToggle = e => {
+    if (sessionStorage.getItem('token')) {
+      const selectedModalBtn = e.currentTarget.getAttribute('name');
+      setIsToggle(prev => {
+        return {
+          ...prev,
+          [selectedModalBtn]: !prev[selectedModalBtn],
+        };
+      });
+    } else {
+      alert('로그인 해주세요');
+    }
   };
+
   const onModalConfirm = e => {
     e.preventDefault();
-    setFavoriteModal(false);
+    setIsToggle(!isToggle);
+    fetch(`${BASE_URL}products/${id}`)
+      .then(res => res.json())
+      .then(data => {
+        setProductData(data);
+      });
   };
 
   useEffect(() => {
-    fetch('/data/Products.json')
+    fetch(`${BASE_URL}products/${id}`)
       .then(res => res.json())
       .then(data => {
         setProductData(data);
       });
-  }, []);
+  }, [id]);
+
+  const btnClick = e => {
+    e.preventDefault();
+    const selectedBtn = e.currentTarget.getAttribute('name');
+    const selectedSize = e.currentTarget.getAttribute('value');
+
+    setIsClickBtn(prev => {
+      return {
+        ...prev,
+        [selectedBtn]: !prev[selectedBtn],
+      };
+    });
+
+    fetch(`${BASE_URL}products/follow`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: sessionStorage.getItem('token'),
+      },
+      body: JSON.stringify({
+        user_id: 2,
+        product_id: Number(productId),
+        size_id: Number(selectedSize),
+      }),
+    }).then(res => res.json());
+  };
+
+  const ModalListObj = {
+    1: (
+      <FavoriteModal
+        title="관심 상품 추가"
+        confirmText="확인"
+        productName={productName}
+        productImg={productImg}
+        productSizes={SIZE_INFO}
+        productId={productId}
+        onConfirm={onModalConfirm}
+        btnClick={btnClick}
+        isClickBtn={isClickBtn}
+      />
+    ),
+    2: (
+      <MarketPriceModal
+        title="시세"
+        productName={productName}
+        productImg={productImg}
+        productSizes={SIZE_INFO}
+        productId={productId}
+        onConfirm={onModalConfirm}
+      />
+    ),
+  };
```

Comment:
```suggestion
  const ModalListObj = {
    1: (
      <FavoriteModal
        title="관심 상품 추가"
        confirmText="확인"
        productName={productName}
        productImg={productImg}
        productSizes={SIZE_INFO}
        productId={productId}
        onConfirm={onModalConfirm}
        btnClick={btnClick}
        isClickBtn={isClickBtn}
      />
    ),
    2: (
      <MarketPriceModal
        title="시세"
        productName={productName}
        productImg={productImg}
        productSizes={SIZE_INFO}
        productId={productId}
        onConfirm={onModalConfirm}
      />
    ),
  };
```

1, 2를 key 로 쓰는 것 보다는 의미가 드러나는 key 값을 쓰는게 좋아보입니다!

Reply:
반영완료